packages/zwave-js/src/lib/commandclass/BinarySwitchCC.ts

```diff
@@ -145,6 +122,21 @@ export class BinarySwitchCCAPI extends CCAPI {
 			throwWrongValueType(this.ccId, property, "boolean", typeof value);
 		}
 		await this.set(value);
+
+		// If the command did not fail, assume that it succeeded and update the currentValue accordingly
+		// so UIs have immediate feedback
+		if (this.isSinglecast()) {
+			const valueDB = this.endpoint.getNodeUnsafe()?.valueDB;
+			valueDB?.setValue(
+				getCurrentValueValueId(this.endpoint.index),
+				value,
+			);
+
+			// Verify the current value after a delay
+			// TODO: #1321
+			const duration = undefined as Duration | undefined;
+			this.schedulePoll({ property }, duration?.toMilliseconds() ?? 1000);
+		}
```

Comment:
Given the frequency of this pattern, I wonder if abstracting into a `setAndVerify` abstraction might be a good idea? Something like…

```suggestion
		// Verify the current value after a delay
		// TODO: #1321
		const duration = undefined as Duration | undefined;
		await this.setAndVerify(value, duration?.toMilliseconds() ?? 1000);
```

Reply:
The problem here is that the value we're immediately updating is different for each CC.
I plan on reworking the value definitions, but until then that's not doable